### ISSUE TITLE
Questions about Equation 6.

### ISSUE THREAD

**Issue by jugechengzi** [Wuyxin/DIR-GNN#10]
Thank you very much for providing such a good paper.
But I am a little confused by Equation 6.  As you first get a soft maks socre (M) for each edge, and you select the top-r edges in Equation 6.  Are The selected edges soft values between [0,1] or just binarized {0,1} values?  If they are binarized values, it seems that the gradients will be detached. But if they are continuous values, it will be strange to select top-r of them.

**Comment by Wuyxin** [comment]
Hi, 

Thanks for your interest in our work.

We use soft scores for the edge weights. See https://github.com/Wuyxin/DIR-GNN/issues/7 for some related details.

Continuous values are soft approximation for the DIR objective and are generally adopted in works like gaph topk pooling or differentiable pooling. By "strange", do you mean it is not binary? For binarization, one can use [gumbel softmax](https://pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html) to make the weight as binary values while they are differentiable, while we didn't find it works very well in the practice.

**Comment by jugechengzi** [comment]
Thank you very much, I get it!
